torchvision/models/video/mvit.py

```diff
@@ -0,0 +1,640 @@
+        temporal_size=temporal_size,
+        block_setting=block_setting,
+        residual_pool=kwargs.pop("residual_pool", False),
+        stochastic_depth_prob=stochastic_depth_prob,
+        **kwargs,
+    )
+
+    if weights is not None:
+        model.load_state_dict(weights.get_state_dict(progress=progress))
+
+    return model
+
+
+class MViT_V1_B_Weights(WeightsEnum):
+    KINETICS400_V1 = Weights(
+        url="https://download.pytorch.org/models/mvit_v1_b-dbeb1030.pth",
+        transforms=partial(
+            VideoClassification,
+            crop_size=(224, 224),
+            resize_size=(256,),
+            mean=(0.45, 0.45, 0.45),
+            std=(0.225, 0.225, 0.225),
+        ),
+        meta={
+            "min_size": (224, 224),
+            "min_temporal_size": 16,
+            "categories": _KINETICS400_CATEGORIES,
+            "recipe": "https://github.com/facebookresearch/pytorchvideo/blob/main/docs/source/model_zoo.md",
+            "_docs": """These weights support 16-frame clip inputs and were ported from the paper.""",
+            "num_params": 36610672,
+            "_metrics": {
+                "Kinetics-400": {
+                    "acc@1": 78.47,
+                    "acc@5": 93.65,
+                }
+            },
+        },
+    )
+    DEFAULT = KINETICS400_V1
+
+
+def mvit_v1_b(*, weights: Optional[MViT_V1_B_Weights] = None, progress: bool = True, **kwargs: Any) -> MViT:
+    """
+    Constructs a base MViTV1 architecture from
+    `Multiscale Vision Transformers <https://arxiv.org/abs/2104.11227>`__.
+
+    Args:
+        weights (:class:`~torchvision.models.video.MViT_V1_B_Weights`, optional): The
+            pretrained weights to use. See
+            :class:`~torchvision.models.video.MViT_V1_B_Weights` below for
+            more details, and possible values. By default, no pre-trained
+            weights are used.
+        progress (bool, optional): If True, displays a progress bar of the
+            download to stderr. Default is True.
+        **kwargs: parameters passed to the ``torchvision.models.video.MViT``
+            base class. Please refer to the `source code
+            <https://github.com/pytorch/vision/blob/main/torchvision/models/video/mvit.py>`_
+            for more details about this class.
+
+    .. autoclass:: torchvision.models.video.MViT_V1_B_Weights
+        :members:
+    """
+    weights = MViT_V1_B_Weights.verify(weights)
+
+    block_setting = [
+        MSBlockConfig(
+            num_heads=1,
+            input_channels=96,
+            output_channels=192,
+            kernel_q=[],
+            kernel_kv=[3, 3, 3],
+            stride_q=[],
+            stride_kv=[1, 8, 8],
+        ),
+        MSBlockConfig(
+            num_heads=2,
+            input_channels=192,
+            output_channels=192,
+            kernel_q=[3, 3, 3],
+            kernel_kv=[3, 3, 3],
+            stride_q=[1, 2, 2],
+            stride_kv=[1, 4, 4],
+        ),
```

Comment:
Nit: I would love to ideally compress this to a single line per block to avoid adding almost 150 lines of code in the implementation and make it resemble more of a table.

I debated wether shortening the names of the variables would do the trick but decided against it because the names should follow what the implementation of the `MViT` class uses internally (keeping in mind that the same names are used on PyTorch Video's implementation). There are 2 options for that:
1. Find a way to turn off the linter for this specific lines. The major drawback here is that any workaround might cause issues on FBcode of things.
2. Move the values in a shorter list and create the `MSBlockConfig()` records within a loop. The list would look like:

```python
config = [
    [1, 96, 192, [], [3, 3, 3], [], [1, 8, 8]],
    [2, 192, 192, [3, 3, 3], [3, 3, 3], [1, 2, 2], [1, 4, 4]],
    ....
]

block_setting = []
for c in config:
        block_setting.append(MSBlockConfig(
            num_heads=c[0],
            input_channels=c[1],
            output_channels=c[2],
            kernel_q=c[3],
            kernel_kv=c[4],
            stride_q=c[5],
            stride_kv=c[6],
        ))
```

Whether this improves readability is arguable because now the original list is error prone and hard to read.

What is a real problem here is that each variant will add 100-200 lines of code, so the moment we start adding more v1 and v2 variants this file will get into the ~2000 lines range. Thoughts?

cc @NicolasHug @jdsgomes @YosuaMichael

Reply:
I am in favour of the option2, perhaps with a dictionary instead of just a list:

```python
config = {
  "num_heads" = [1, 2, 4, ...],
  "input_channels" = [192, 192, ...],
  "output_channels" = [...],
  "kernel_q" = [...],
  "kernel_kv" = [...],
  "stride_q" = [...],
  "stride_kv" = [...],
}

```

This way will be more informative when someone looks at the config

Reply:
This would make it perfectly aligned with the PyTorch Video configuration. I'll do that, thanks @jdsgomes